### PR TITLE
Improve upgrade guide and release process docs

### LINKED ISSUE #12543: Clarify our deprecation policy and loosen it for features in "Extending Wagtail"
### Pertinent section of the Wagtail docs

We have two sections describing our deprecation policy:
- https://docs.wagtail.org/en/stable/releases/upgrading.html#deprecation-policy
- https://docs.wagtail.org/en/stable/contributing/release_process.html#deprecation-policy

and both of them are outdated 😬 

### Details

Our docs state that we follow a "loose form of semantic versioning". That line has existed even since v1.x days. We moved on to follow SemVer more closely in recent releases since 3.0 as discussed in https://github.com/wagtail/wagtail/discussions/7732#discussioncomment-2207842. However, it looks like this is still undocumented – despite I'm pretty sure we mentioned it somewhere, maybe in a past What's New in Wagtail episode.

With our new deprecation policy, deprecated features are only removed in the next major release (`A.0`). Following our previous policy, a deprecation will last for _at least_ two releases: the release where the feature is deprecated, and the subsequent release. The feature will raise a warning for those releases. The difference is that the feature may not be removed outright in the release after that – it may stick around until whenever the next major release is decided.

Our major releases are not regularly scheduled, which means future version numbers are provisional. A major version bump is tentatively decided when we have accumulated enough deprecations that it makes sense to remove them.

For this reason, if it turns out a deprecation cannot go through the minimum cycle of two releases before the next major release, we bump the deprecation warning to the next release. For example: any features deprecated in 5.2 were not immediately removed in 6.0. Instead, they will be removed in 7.0 (whenever that will be).

### Working on this

I'll work on this, along with clarifying which parts are subject to the policy. For example, anything under the "Extending Wagtail" section of the docs is not subject to the deprecation policy. We'll try to add deprecation warnings instead of outright removing them, but they may be removed in a non-major release if keeping the old code is not deemed to be worth it.

### PR DESCRIPTION
**Pull request by laymonage**
Fixes #12543 per #12546.

### CODE REVIEW

**Review by gasman — APPROVED** [review]
Nice improvements here - all looks good to me!

File: docs/contributing/release_process.md [hunk 1]
```diff
@@ -41,38 +44,74 @@ unless this is impossible for security reasons or to prevent data loss.
...
 be "yes."
 
+A feature release will usually stop receiving patch release updates when the next feature release comes out.
+
 **Long-term support release**
 
 Certain feature releases will be designated as long-term support (LTS)
 releases. These releases will get security and data loss fixes applied for
-a guaranteed period of time, typically six months.
+a guaranteed period of time. Typically, a Long Term Support release will happen
```
Comment (lb-) [inline docs/contributing/release_process.md:53]: Nitpicky but we use `Long-term support release` twice in the above text and your change has `Long Term Support release`.

Might be good to be consistent, whichever way we go.

**Review by lb- — APPROVED** [review]
I know it's been approved, great work here. This reflects my understanding of the process also. 

We may want to call out these upgrade policy changes a bit louder in the release notes with a bit of a summary. Especially the extending parts being potentially less stable. 

I know it's been like this for a while but best to over communicate for this stuff. 

Also, if possible, it would be great if you could run Prettier (force format) over this. It'll clean up some of the white space and indents on lists.

**Review by laymonage — COMMENTED** [review]
Thanks both for the reviews. I added another commit for some minor tweaks.

> Also, if possible, it would be great if you could run Prettier (force format) over this. It'll clean up some of the white space and indents on lists.

@lb- Sure, just did now. Though we have the following line in our `.prettierignore`, so I'm not sure whether we're expected to run Prettier for docs or not:

https://github.com/wagtail/wagtail/blob/246f3c7eb542be2081556bf5334bc22256776bf4/.prettierignore#L10-L11

**Comment by lb-** [comment]
We don't enforce Prettier in docs, it's just nice to have it cleaned up every now and then.

**Pull request merged**